File: OpenLive-iOS-Objective-C/OpenLive/ProfileCell.m
//
//  ProfileCell.m
//  OpenLive
//
//  Created by GongYuhua on 2016/9/12.
//  Copyright © 2016年 Agora. All rights reserved.
//

#import "ProfileCell.h"

@interface ProfileCell()
@property (weak, nonatomic) IBOutlet UILabel *resLabel;
@property (weak, nonatomic) IBOutlet UILabel *frameLabel;
@end

@implementation ProfileCell

- (void)updateWithProfile:(CGSize)profile isSelected:(BOOL)isSelected {
    self.resLabel.text = [self resolutionOfProfile:profile];
    self.frameLabel.text = [self fpsOfProfile:profile];
    self.backgroundColor = isSelected ? [UIColor colorWithRed:0 green:0 blue:0.5 alpha:0.3] : [UIColor whiteColor];
}

- (NSString *)resolutionOfProfile:(CGSize)profile {
    return [NSString stringWithFormat:@"%d×%d", (int)profile.width, (int)profile.height];
}

- (NSString *)fpsOfProfile:(CGSize)profile {
    return @"24";
}
@end

File: OpenLive-iOS/OpenLive/ProfileCell.swift
//
//  ProfileCell.swift
//  OpenLive
//
//  Created by GongYuhua on 6/26/16.
//  Copyright © 2016 Agora. All rights reserved.
//

import UIKit
import AgoraRtcEngineKit

class ProfileCell: UITableViewCell {
    
    @IBOutlet weak var resLabel: UILabel!
    @IBOutlet weak var frameLabel: UILabel!
    
    func update(withProfile profile: AgoraVideoProfile, isSelected: Bool) {
        resLabel.text = profile.resolution()
        frameLabel.text = profile.fps()
        backgroundColor = isSelected ? UIColor(red: 0, green: 0, blue: 0.5, alpha: 0.3) : UIColor.white
    }
}

private extension AgoraVideoProfile {
    func resolution() -> String? {
        switch self {
        case .landscape120P: return "160×120"
        case .landscape180P: return "320×180"
        case .landscape240P: return "320×240"
        case .landscape360P: return "640×360"
        case .landscape480P: return "640×480"
        case .landscape720P: return "1280×720"
        default: return nil
        }
    }
    
    func fps() -> String? {
        return "15"
    }
}
